File: frontend/src/api/internship.ts
import { http } from '@/utils/request'

/**
 * 实习经历接口类型定义
 */
export interface Internship {
  id?: number
  userId?: number
  companyName: string
  position: string
  department?: string
  startDate: string // YYYY-MM-DD
  endDate?: string // YYYY-MM-DD, 空表示至今
  description?: string
  createTime?: string
  updateTime?: string
}

/**
 * 根据用户ID获取实习经历列表
 */
export function getInternshipByUserIdApi(userId: number) {
  return http.get<Internship[]>(`/internship/user/${userId}`)
}

/**
 * 根据ID获取实习经历详情
 */
export function getInternshipByIdApi(id: number) {
  return http.get<Internship>(`/internship/${id}`)
}

/**
 * 创建实习经历
 */
export function createInternshipApi(data: Internship) {
  return http.post<void>('/internship', data)
}

/**
 * 更新实习经历
 */
export function updateInternshipApi(data: Internship) {
  return http.put<void>('/internship', data)
}

/**
 * 删除实习经历
 */
export function deleteInternshipApi(id: number) {
  return http.delete<void>(`/internship/${id}`)
}

File: backend/src/main/java/com/teaching/employment/entity/Internship.java
package com.teaching.employment.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonFormat;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 实习经历实体类
 *
 * @author Teaching Employment Platform Team
 * @since 2026-01-18
 */
@Data
@TableName("t_internship")
@ApiModel(description = "实习经历信息")
public class Internship implements Serializable {

    private static final long serialVersionUID = 1L;

    @ApiModelProperty(value = "主键ID")
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    @ApiModelProperty(value = "用户ID")
    private Long userId;

    @ApiModelProperty(value = "公司名称")
    private String companyName;

    @ApiModelProperty(value = "职位")
    private String position;

    @ApiModelProperty(value = "部门")
    private String department;

    @ApiModelProperty(value = "开始时间")
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate startDate;

    @ApiModelProperty(value = "结束时间(空表示至今)")
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate endDate;

    @ApiModelProperty(value = "工作描述/职责")
    private String description;

    @ApiModelProperty(value = "创建时间")
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @ApiModelProperty(value = "更新时间")
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}
